dbx/commands/sync.py

```diff
@@ -0,0 +1,384 @@
+def check(profile: str):
+    """
+    Check if configuration for Databricks is valid.
+    """
+    # This will fail if the API token doesn't work.
+    get_databricks_config(profile)
+
+    click.echo(click.style("Databricks API token is properly configured and working", fg="green"))
+
+
+def handle_source(source: str = None) -> str:
+    """Determine the source directory to sync from.  If the source directory is not specified
+    then it will check if the current directory has a .git subdirectory, implying that the current
+    directory should be the source.
+
+    Args:
+        source (str): source directory to use, or None to try automatically determining
+
+    Raises:
+        click.UsageError: When the source directory is unspecified and can't be automatically determined.
+
+    Returns:
+        str: source directory to sync from
+    """
+    if not source:
+        # If this is being run from the base of a git repo, then they probably want to use
+        # the repo as the source.
+        if os.path.exists(".git"):
+            source = "."
+        else:
+            raise click.UsageError("Must specify source directory using --source")
+
+    source = os.path.abspath(source)
+
+    dbx_echo(f"Syncing from {source}")
+
+    return source
+
+
+def common_options(f):
+    f = click.option("--profile", type=str, help="The databricks-cli profile to use for accessing the API token")(f)
+    f = click.option("--source", "-s", type=click.Path(exists=True, dir_okay=True, file_okay=False), required=False)(f)
+    f = click.option("--full-sync", is_flag=True, help="Copy everything from the source on the initial sync")(f)
+    f = click.option("--dry-run", is_flag=True)(f)
+    f = click.option(
+        "--include",
+        "-i",
+        "include_dirs",
+        multiple=True,
+        type=str,
+        required=False,
+        help="Directory under the source directory to sync",
+    )(f)
+    f = click.option("--exclude", "-e", "exclude_dirs", type=str, multiple=True)(f)
+    f = click.option(
+        "--include-pattern",
+        "-ip",
+        "include_patterns",
+        multiple=True,
+        type=str,
+        required=False,
+        help="Pattern under the source directory to sync",
+    )(f)
+    f = click.option(
+        "--allow-delete-unmatched/--disallow-delete-unmatched",
+        "delete_unmatched_option",
+        callback=validate_allow_unmatched,
+        type=bool,
+        default=None,
+        help="How to handle files/directories that would be deleted in remote because they don't match a filter",
+    )(f)
+    f = click.option("--exclude-pattern", "-ep", "exclude_patterns", type=str, multiple=True)(f)
+    f = click.option("--watch/--no-watch", is_flag=True, default=True)(f)
+    return f
+
+
+@cli.command()
+@common_options
+@click.option("--dest", "-d", "dest_path", type=str, required=False)
+@click.option("--user", "-u", "user_name", type=str)
```

Comment:
I saw this earlier looking in the docs and thought it'd be great to use but noticed it says "This feature is in Public Preview" and:

```
Your Databricks account must have the Premium plan (or, for customers who subscribed to Databricks before March 3, 2020, the Operational Security package).
```

Is this no longer the case?

Reply:
I think what I can probably do though is try to use the API and use the username if we can get it, but if the API isn't available then it can fall back to requiring the user provide it.

Reply:
I updated the code to try using scim if possible but allow specifying with `--user` when it's not available.

Reply:
@matthayes SCIM API is "preview" for two years, but everyone is using it in production already :)